src/main/java/com/google/firebase/database/FirebaseDatabase.java

```diff
@@ -112,14 +121,18 @@ public static synchronized FirebaseDatabase getInstance(FirebaseApp app, String
     if (service == null) {
       service = ImplFirebaseTrampolines.addService(app, new FirebaseDatabaseService());
     }
-
-    DatabaseInstances dbInstances = service.getInstance();
     if (url == null || url.isEmpty()) {
       throw new DatabaseException(
           "Failed to get FirebaseDatabase instance: Specify DatabaseURL within "
               + "FirebaseApp or from your getInstance() call.");
     }
-
+    boolean connectingToEmulator = false;
+    String possibleEmulatorUrl = EmulatorHelper
+        .overwriteDatabaseUrlWithEmulatorHost(url, EmulatorHelper.getEmulatorHostFromEnv());
+    if (!Strings.isNullOrEmpty(possibleEmulatorUrl)) {
+      url = possibleEmulatorUrl;
+      connectingToEmulator = true;
+    }
     ParsedUrl parsedUrl = Utilities.parseUrl(url);
     if (!parsedUrl.path.isEmpty()) {
```

Comment:
In `getInstance()`, we don't allow any custom paths. I think with that in mind we might not need to merge `url` and the environment variable? We would only need to return `repoInfo` based on the emulator host.

Reply:
FYI Python Admin SDK will strip out the path part from the URL.

Reply:
Let's maintain backwards compatibility here and keep throwing the exception that was being thrown earlier.